### ISSUE TITLE
Append the scheme part of the HTTP client URL based on the client configurations

### ISSUE THREAD

**Issue by ldclakmal** [ballerina-platform/ballerina-standard-library#2816]
**Description:**
This issue is created to track the improvement of appending the scheme part to HTTP client URL based on the client configurations.

**Scenario - 1:** Should append `http://` to the URL

```ballerina
http:Client usersEP = check new("localhost:9090");
```

**Scenario - 2:** Should append `https://` to the URL

```ballerina
http:Client usersEP = check new("localhost:9090",
    secureSocket = {
        cert: "./resources/public.crt"
    }
);
```

**Scenario - 3:** Should append `https://` to the URL

```ballerina
http:Client usersEP = check new("localhost:9090",
    auth = {
        username: "alice",
        password: "123"
    }
);
```

**Scenario - 4:** Should append `https://` to the URL

```ballerina
http:Client usersEP = check new("localhost:9090",
    auth = {
        username: "alice",
        password: "123"
    },
    secureSocket = {
        cert: "./resources/public.crt"
    }
);
```

**Scenario - 5:** Should be able to override the URL protocol as `http://` for any of the above scenarios.

**NOTE:** But there will be a runtime warning if the protocol is overridden as `http://` for scenario 2, 3 & 4.

```ballerina
http:Client usersEP = check new("http://localhost:9090",
    auth = {
        username: "alice",
        password: "123"
    },
    secureSocket = {
        cert: "./resources/public.crt"
    }
);
```

```shell
warning: [ballerina/http] HTTPS is recommended but using HTTP
```

#### In summary:
- If the scheme part is not provided & no security related configurations added, `http://` scheme will be added.
- If the scheme part is not provided & one or more security related configurations are added, `https://` scheme will be added.
- If the scheme part is added as `http://` & one or more security related configurations are also added, there will be a warning.

Furthermore, this support should be extended to the client configurations of JWT and OAuth2 modules as well.

**Comment by ldclakmal** [comment]
Fixed with the above PRs.
